src/gedi_subset/subset.py

```diff
@@ -154,37 +156,39 @@ def subset_granule(props: SubsetGranuleProps) -> IOResultE[Maybe[str]]:
     GeoParquet file.
     """
 
-    inpath = download_granule(props.granule, str(props.output_dir))
+    if not (inpath := props.granule.getDownloadUrl()):
+        granule_ur = props.granule["Granule"]["GranuleUR"]
+        logger.warning(f"Skipping granule {granule_ur} [no download URL]")
+        return IOSuccess(Nothing)
 
     logger.debug(f"Subsetting {inpath}")
+    fs = props.fs or s3fs.S3FileSystem()
```

Comment:
In SubsetGranuleProps is fs being initialized as none? Are there instances when props.fs is defined?

Reply:
This is for testing purposes. The unit test passes in a value.